sre_yield/__init__.py

```diff
@@ -528,7 +541,13 @@ def check_anchor_state(self, matcher, arguments):
 
     def __init__(self, pattern, flags=0, charset=CHARSET, max_count=None):
         # If the RE module cannot compile it, we give up quickly
-        self.matcher = re.compile(r"(?:%s)\Z" % pattern, flags)
+        if isinstance(pattern, (tuple, sre_parse.SubPattern)):
+            pattern = _finalise_parsed_pattern(pattern)
+        else:
+            if not pattern.endswith(r")\Z"):
```

Comment:
At some point soon I'm fine dropping py2 compatibility.  Since 3.4 there has been a fullmatch method, which we can use without appending `\Z`

Reply:
If we do that, can `_finalise_parsed_pattern` go away because it's a noop?

Reply:
I'm fine with py2 being dropped.  It would be better if you did it, as I am sure you know all the bits of py2 specific logic which can be removed.

Reply:
I did so in master, the tests now run on 3.6+

Reply:
Nod. Rebasing